sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/SpecializedGettersReader.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions;
+
+import org.apache.spark.sql.types.*;
+
+public class SpecializedGettersReader {
+  private final boolean handleNull;
+  private final boolean handleUserDefinedType;
+
+  public SpecializedGettersReader(boolean handleNull, boolean handleUserDefinedType) {
+    this.handleNull = handleNull;
+    this.handleUserDefinedType = handleUserDefinedType;
+  }
+
+  public Object read(SpecializedGetters obj, int ordinal, DataType dataType) {
```

Comment:
We could even make the class as utility class and no need to make object: just change this to static, and add two boolean parameters. If we prefer not to have object even it's only one for each class, I'll make a change.

Reply:
I tend to agree, a new object isn't needed.
It could plausibly live in the SpecializedGetters interface in Java 8, but that might be weird.

Reply:
Yes, I tried to add it to interface via default method, but looked weird so decided to have utility class for this.